### ISSUE TITLE
line length linter gives incorrect warnings with UTF-8 characters

### ISSUE THREAD

**Issue by tomjemmett** [r-lib/lintr#1422]
I've noticed that lintr fails on a file that contains a line like this:

```
# ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

(a #, followed by a space, followed by [U+2500](https://www.compart.com/en/unicode/U+2500), for a line length of 120).

I run something like

``` r
lintr::lint("filename.R", linters = c(line_length_linter(120)))
```

but get

```
filename.R:1:121: style: [line_length_linter] Lines should not be more than 120 characters.
# ââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââ
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

setting the linter to 356 works, 355 fails. This sort of makes sense if the text is not being loaded as UTF-8 (1 char + 1 char + 118 * 3 char = 356).

Digging into this, is seems like the source of the issue may be in `get_source_expressions()` and the call to `base::srcfile`. On my system I have `getOption("encoding") == "native.enc"` (I am pretty certain that this is the default shipped with R as I have never had to change this previously), but setting `options("encoding" = "UTF-8")` causes the issue to go away.

I've tested this both on Windows and Mac OS, both with same results, using R 4.2.0 and lintr 3.0.0.9000, below is a subset of my session info. This feels like it's just a general text encoding issue so happy for this to be classed as out-of-scope.

```
 setting  value
 version  R version 4.2.0 (2022-04-22)
 os       macOS Big Sur/Monterey 10.16
 system   x86_64, darwin17.0
 ui       X11
 language (EN)
 collate  en_GB.UTF-8
 ctype    en_GB.UTF-8
 tz       Europe/London
 date     2022-06-19
 pandoc   2.18 @ /usr/local/bin/pandoc
```

**Comment by AshesITR** [comment]
I can't reproduce. Does this code reproduce the issue for you?
What R console are you using?

```r
tmp <- withr::local_tempfile()
writeLines(paste0("# ", strrep("\U2500", 118)), tmp)
lintr::lint(tmp, line_length_linter(120L))
```

```
 setting  value
 version  R version 4.2.0 (2022-04-22)
 os       Arch Linux
 system   x86_64, linux-gnu
 ui       RStudio
 language (EN)
 collate  en_DK.UTF-8
 ctype    en_DK.UTF-8
 tz       Europe/Berlin
 date     2022-06-19
 rstudio  1.3.1056  (desktop)
 pandoc   2.17.1.1 @ /usr/bin/ (via rmarkdown)
```

**Comment by tomjemmett** [comment]
I think I've tracked it down to the wrong value in .lintr file which for some reason had encoding overidden. Setting this to UTF-8 sorts the issue!
